public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -28,17 +30,32 @@ export const RestoreActivitiesPanel: React.FC<RestoreActivitiesPanelProps> = ({
   const [flyout, setFlyout] = useState(false);
 
   useEffect(() => {
-    context?.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
-    getRestoreStatus();
-  }, []);
+    context!.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
+
+    let getStatusInterval: ReturnType<typeof setInterval>;
+
+    if (stage.slice(0, 4) !== "Done") {
+      getStatusInterval = setInterval(() => {
+        getRestoreStatus();
+      }, 1000);
+      return () => {
+        clearInterval(getStatusInterval)
+      }
+    }
+  }, [stage]);
 
   const getRestoreStatus = async () => {
-    if (stage.indexOf("DONE") >= 0) {
+    if (!restoreStartRef) {
```

Comment:
Not sure why do we need this. This seems to be set after user click the restore.